src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFCD.cs

```diff
@@ -58,19 +58,26 @@ private RegexFCD(ValueListBuilder<int> intStack)
             // Create/rent buffers
             Span<int> intSpan = stackalloc int[StackBufferSize];
             var intStack = new ValueListBuilder<int>(intSpan);
-
-            RegexFCD s = new RegexFCD(intStack);
-            RegexFC fc = s.RegexFCFromRegexTree(t);
-
-            // Return rented buffers
+            RegexFC[] fcSpan = ArrayPool<RegexFC>.Shared.Rent(StackBufferSize);
+            var fcStack = new ValueListBuilder<RegexFC>(fcSpan, clearArray: true);
```

Comment:
why is clearArray important? we almost never clear on returning.

Reply:
the clearArray here is wrong as I do it manually some lines below but in general clearArray is important if you store managed objects with references to other objects (in our case RegexFC --> RegexCharClass). We usually use ArrayPool in combination with primitive types like int & char.

Reply:
I accidently merged two commits, don't wonder :)